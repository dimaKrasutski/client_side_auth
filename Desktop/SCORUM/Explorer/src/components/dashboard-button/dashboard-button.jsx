import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Button from 'reactstrap/lib/Button';

export const DashboardButton = ({ text, onClick }) => (
  <Button outline size="xs" className="float-right mt-0_5" onClick={onClick}>
    <FormattedMessage id={text} />
  </Button>
);

DashboardButton.propTypes = {
  text: PropTypes.string.isRequired,
  onClick: PropTypes.func,
};

DashboardButton.defaultProps = {
  onClick: null,
};
