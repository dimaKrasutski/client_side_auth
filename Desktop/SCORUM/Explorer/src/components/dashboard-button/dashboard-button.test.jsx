import React from 'react';
import { DashboardButton } from './dashboard-button';

describe('<DashboardButton />', () => {
  test('should match snapshot', () => {
    const wrapper = shallow(<DashboardButton text="test" onClick={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });
});
