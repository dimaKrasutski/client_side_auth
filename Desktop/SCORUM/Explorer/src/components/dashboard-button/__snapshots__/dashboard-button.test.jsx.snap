// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DashboardButton /> should match snapshot 1`] = `
<Button
  className="float-right mt-0_5"
  color="secondary"
  onClick={[MockFunction]}
  outline={true}
  size="xs"
  tag="button"
>
  <FormattedMessage
    id="test"
    values={Object {}}
  />
</Button>
`;
